pkgs/by-name/u-/u-config-unwrapped/package.nix

```diff
@@ -0,0 +1,80 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  installShellFiles,
+  versionCheckHook,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "u-config";
+  version = "0.33.3";
+
+  src = fetchFromGitHub {
+    owner = "skeeto";
+    repo = "u-config";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-2chZwS8aC7mbPJwsf5tju2ZNZNda650qT+ARjNJ2k2g=";
+  };
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
+  dontConfigure = true;
+
+  buildPhase = ''
+    runHook preBuild
+
+    # Every pkg-config implementation needs distro-specific configuration for where the .pc
+    # files are installed, hence the default Makefile does not contain target for building
+    # and installation, only for development. Instead of using it, we build the u-config via
+    # manual CC invocation. Since on NixOS there is no global pc-path we set it to /var/empty
+    # and rely on pkg-config-wrapper to provide the correct search path at runtime.
```

Comment:
Just a heads up: `pc_path`, `pc_system_includedirs` & `pc_system_libdirs` virtual variables are expected to output the ["compiled in" paths and not change due to runtime configuration](https://github.com/skeeto/u-config/commit/ccd27ff688208c19bb2fbb8869876a295350c8af). Some build systems or whatnot may try to query it.

But I'm not a nix user though, so I cannot comment on whether this is fine or not. If it's fine to output `/var/empty[/...]` for these virtual variables then you can ignore this.

Reply:
> I was successful but only after applying [skeeto/u-config@62c88c0](https://github.com/skeeto/u-config/commit/62c88c0803f05ba33deb6151a20ca5d6b61598b6). I don't know enough about bootstrapping to understand this but similar patch is currently applied to `pkg-config` in nixpkgs and it [was causing some problems](https://github.com/NixOS/nixpkgs/issues/334195#issuecomment-2428781605). I see that it's reverted in `u-config` now.

The [gentoo ebuild for glib](https://github.com/gentoo/gentoo/blob/824f8f537280e5419d64edeeb1f1112ff7e3c4a7/dev-libs/glib/glib-2.84.0.ebuild#L281-L288) does some shenanigans with `PKG_CONFIG_*` vars to make it work without patching pkgconf/pkg-config. Maybe you could employ similar workaround in nixos's glib build too?